packages/base/src/asset-registries/Illustrations.js

```diff
@@ -13,18 +15,39 @@ const registerIllustration = (name, { dialogSvg, sceneSvg, spotSvg, set, title,
 	});
 };
 
+const registerIllustrationLoader = async (illustrationName, loader) => {
+	loaders.set(illustrationName, loader);
+};
+
+const _loadIllustrationOnce = async illustrationName => {
+	if (!illustrationPromises.has(illustrationName)) {
+		if (!loaders.has(illustrationName)) {
+			throw new Error(`No loader registered for the ${illustrationName} illustrations collection. Probably you forgot to import the "AllIllustrations.js" module for the respective package.`);
+		}
+		const loadIllustrations = loaders.get(illustrationName);
+		illustrationPromises.set(illustrationName, loadIllustrations(illustrationName));
+	}
+	return illustrationPromises.get(illustrationName);
+};
+
 const getIllustrationDataSync = nameProp => {
 	let set = "fiori";
 
 	if (nameProp.startsWith("Tnt")) {
 		set = "tnt";
 		nameProp = nameProp.replace(/^Tnt/, "");
 	}
+	return registry.get(`${set}/${nameProp}`);
+};
 
-	return registry.get(`${set}/${nameProp}`) || ILLUSTRATION_NOT_FOUND;
+const getIllustrationData = async illustrationName => {
+	await _loadIllustrationOnce(illustrationName);
+	return registry.get(`${illustrationName}`) || ILLUSTRATION_NOT_FOUND;
```

Comment:
This looks wrong to me: "return registry.get(`${illustrationName}`)"

If you look at `getIllustrationDataSync`, there you get the illustration data with:
"return registry.get(`${set}/${nameProp}`);"

So, one of them should be wrong, and I think it's code in the async method - `getIllustrationData`.
You need to see how you set the data into the registry and then you will know the way to get it.

It works by chance, because the component re-renders and then it gets the data via the sync method -  `getIllustrationDataSync`.